### PR TITLE
BUG FIX: Configured retry policy attempts are multiplied by a factor of 3 for OpenAI & several other providers

### PR DESCRIPTION
**Pull request by dbczumar**
## Title

BUG FIX: Configured retry policy attempts are multiplied by a factor of 3 for OpenAI

## Type

🐛 Bug Fix

## Changes

There's logic in the OpenAI provider (and several other providers) sets `max_retries = 2` (for a total of 3 request attempts) if `max_retries` is not specified in the call to `completion()`: https://github.com/BerriAI/litellm/blob/1ef311343c93fdccec0a0e07a3079febcb9c433e/litellm/llms/OpenAI/openai.py#L546.

As a result, if a user specifies a retry policy, e.g. with `RateLimitErrorRetries=3`, *six (6)* attempts are made.

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locally

Ran a DSPy test case that reproduces the issue before and after the change.

* Here's DSPy's usage of `retry_policy`: https://github.com/stanfordnlp/dspy/pull/1866/files#diff-c136e37e95b0db7bd8abc41012e4c91f9dbbbf14af7b1c44c71e8ada21e6bb3eR233-R293

* Here's the test case: https://github.com/stanfordnlp/dspy/pull/1866/files#diff-8ec5c2977033edfed9947e0d4cc92fdee332394772b9df13450653685194ec95R28-R61

**Before**

```
FAILED test_lm.py::test_lm_chat_calls_are_retried_for_expected_failures[429-RateLimitError-1] - AssertionError: assert 6 == (1 + 1)
FAILED test_lm.py::test_lm_chat_calls_are_retried_for_expected_failures[504-Timeout-1] - AssertionError: assert 6 == (1 + 1)
```

Screenshot 2024-12-09 at 11 52 41 PM

**After**

Screenshot 2024-12-09 at 11 52 54 PM

### CODE REVIEW

File: litellm/utils.py [hunk 1]
```diff
@@ -930,6 +937,8 @@ def wrapper(*args, **kwargs):  # noqa: PLR0915
                     kwargs["retry_policy"] = (
                         reset_retry_policy()
                     )  # prevent infinite loops
+                    kwargs["max_retries"] = 0
+                    kwargs["num_retries"] = 0
```
Comment (dbczumar) [inline litellm/utils.py:941]: https://github.com/BerriAI/litellm/blob/1ef311343c93fdccec0a0e07a3079febcb9c433e/litellm/llms/OpenAI/openai.py#L546 appears to continually reset `max_retries` to `2` via the `pop()` logic every time the OpenAI provider is called, so we need to reset here after every iteration as well

**Comment by dbczumar** [comment]
@krrishdholakia Can you take a look at this PR? This is probably a fairly hacky attempt at a solution. Regardless, would love to get a fix for this issue (else all retry policy configs are tripled) released with 1.54.2 in the next few days, if possible...

In case it helps in some way, I had observed this issue & accounted for it in my original PR - https://github.com/BerriAI/litellm/pull/6916/files#diff-9661b0d8f1d9f48433f2323d1102e963d3098f667f7a25caea79f16fa282f6ddR689-R703.

cc @okhat

File: litellm/utils.py (hunk 1 above)
Comment (krrishdholakia) [reply litellm/utils.py:941]: @dbczumar the max retries is set to this to handle the user/assistant alternating scenario for openai-compatible routes like Mistral (see the try except block)

Comment (ghost) [reply litellm/utils.py:941]: got it - so it looks like you want to disable the openai max retries param so we don't pass that to their client sdk

**Pull request closed without merging**